### ISSUE TITLE
Browse mode problems with slow/inappropriate focus changes in browsers

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2039]
**Reported by parham on 2012-01-11 12:09**
NVDA is almost always snappy when visiting most websites. However, when the website in question has a large amount of Javascript running, NVDA slows down considerably and scrolling down the page becomes hard (I.E. the focus jumps around and there's a delay between pressing an arrow key and getting a response). The responsiveness of NVDA should improve in these applications as well.
**Blocked by #5411**
**Blocking #1544, #4719, #5032, #5359**

**Comment by feerrenrut** [comment]
Setting the priority on this issue to 2. This seems like something that should be fixed to improve confidence in NVDA, and enable users to maximise their efficiency when navigating pages. Sporadic differences in behaviour can be quite off putting, particularly to new users. Bear in mind that this issue may be quite difficult to fix, and will likely need to be tested quite thoroughly to ensure regressions aren't introduced.

**Comment by seancurtis** [comment]
Is there any plans to fix this soon? A customer of ours that uses NVDA as their screen reader raised a bug with us relating to our dropdown menus and how bad the keyboard support is. We've built them with accessibility in mind, and the focus issue only occurs once they turn NVDA on :(

I've tested in Narrator and Voiceover but they work perfectly. Could you let us know if there are plans to fix this bug?

**Comment by cblouch** [comment]
Just ran into a similar reproducible situation on Firefox and Chrome on Windows 10. I have two containers, A and B, with focusable elements in them. A has display:none, B has display:block. Focus is on a DIV in B. If I hit enter on the DIV a script hides B, unhides A and moves focus to a DIV in A. NVDA is silent even though I moved focus to newly displayed content. If I tab away and come back the element is announced. My hack to fix was to wrap the hide script in a setTimeout, forcing the order of operations to show A, focus inside A and then hide B. Changing the order without the setTimeout also fails. This pattern of hiding something with focus in it, unhiding something else and moving focus there is pretty common. JAWS and VoiceOver on various browser handle this just fine without the timer hack. Maybe hiding a container with a focused element puts NVDA in an odd state where the next focus move is not announced.

Codepen with the failure example is here:
https://codepen.io/cblouch/pen/eVrXoY
Uncomment the NVDA Hack line and comment out the previous line to try with the setTimeout.

**Comment by Adriani90** [comment]
Closing as fixed by #11190.
